components/TokenView/src/TokenView/TokenView.Events.cs

```diff
@@ -0,0 +1,160 @@
+    {
+#if !HAS_UNO
+        if (_tokenViewScrollBackButton != null)
+        {
+            if (e.FinalView.HorizontalOffset < 1)
+            {
+                _tokenViewScrollBackButton.Visibility = Visibility.Collapsed;
+            }
+            else if (e.FinalView.HorizontalOffset > 1)
+            {
+                _tokenViewScrollBackButton.Visibility = Visibility.Visible;
+            }
+        }
+
+        if (_tokenViewScrollForwardButton != null)
+        {
+            if (_tokenViewScroller != null)
+            {
+                if (e.FinalView.HorizontalOffset > _tokenViewScroller.ScrollableWidth - 1)
+                {
+                    _tokenViewScrollForwardButton.Visibility = Visibility.Collapsed;
+                }
+                else if (e.FinalView.HorizontalOffset < _tokenViewScroller.ScrollableWidth - 1)
+                {
+                    _tokenViewScrollForwardButton.Visibility = Visibility.Visible;
+                }
+            }
+        }
+#endif
+    }
+
+    private void ScrollViewer_Loaded(object sender, RoutedEventArgs e)
+    {
+        if (_tokenViewScroller != null)
+        {
+            _tokenViewScroller.Loaded -= ScrollViewer_Loaded;
+        }
+        if (_tokenViewScrollBackButton != null)
+        {
+            _tokenViewScrollBackButton.Click -= ScrollTabBackButton_Click;
+        }
+
+        if (_tokenViewScrollForwardButton != null)
+        {
+            _tokenViewScrollForwardButton.Click -= ScrollTabForwardButton_Click;
+        }
+
+        if (_tokenViewScroller != null)
+        {
+            _tokenViewScroller.ViewChanging += _tokenViewScroller_ViewChanging;
+            _tokenViewScrollBackButton = _tokenViewScroller.FindDescendant(TokenViewScrollBackButtonName) as ButtonBase;
+            _tokenViewScrollForwardButton = _tokenViewScroller.FindDescendant(TokenViewScrollForwardButtonName) as ButtonBase;
+        }
+
+        if (_tokenViewScrollBackButton != null)
+        {
+            _tokenViewScrollBackButton.Click += ScrollTabBackButton_Click;
+        }
+
+        if (_tokenViewScrollForwardButton != null)
+        {
+            _tokenViewScrollForwardButton.Click += ScrollTabForwardButton_Click;
+        }
+
+        UpdateScrollButtonsVisibility();
+    }
+
+    private void Token_Removing(object? sender, TokenItemRemovingEventArgs e)
+    {
+        if (ItemFromContainer(e.TokenItem) is object item)
+        {
+            var args = new TokenItemRemovingEventArgs(item, e.TokenItem);
+            TokenItemRemoving?.Invoke(this, args);
+
+            if (ItemsSource != null)
+            {
+                _removeItemsSourceMethod?.Invoke(ItemsSource, new object[] { item });
+            }
+            else
+            {
+                if (_tokenViewScroller != null)
+                {
+                    _tokenViewScroller.UpdateLayout();
+                }
+                Items.Remove(item);
+            }
+        }
```

Comment:
Found a bug, you also need to check if it's the SelectedItem, and if so set that to null:

![image](https://user-images.githubusercontent.com/24302614/228314623-29984aac-dc04-4f5d-bdbe-13c689d511d2.png)

![image](https://user-images.githubusercontent.com/24302614/228314662-1e0f8ba9-29ce-4b68-85dd-6c1ae91fb6fa.png)

Reply:
Fixed in https://github.com/CommunityToolkit/Labs-Windows/pull/348/commits/877f39393320bf142118900b24a7b5b9673b62de